mqtt/desired_state_client.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) 2023 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+
+package mqtt
+
+import (
+	"fmt"
+
+	"github.com/eclipse-kanto/update-manager/api"
+	"github.com/eclipse-kanto/update-manager/logger"
+
+	pahomqtt "github.com/eclipse/paho.mqtt.golang"
+)
+
+type desiredStateClient struct {
+	*mqttClient
+	domain       string
+	stateHandler api.StateHandler
+}
+
+// NewDesiredStateClient instantiates a new client for triggering MQTT requests.
+func NewDesiredStateClient(domain string, updateAgent api.UpdateAgentClient) api.DesiredStateClient {
+	client := updateAgent.(*updateAgentClient)
+	return &desiredStateClient{
+		mqttClient: &mqttClient{
+			mqttPrefix: domainAsTopic(domain),
+			mqttConfig: client.mqttConfig,
+			pahoClient: client.pahoClient,
+		},
+		domain: domain,
+	}
+}
+
+func (client *desiredStateClient) topic(topicSuffix string) string {
```

Comment:
I guess generating the topics once at the creation of the client will be more efficient than calculating them every time. What do you think?